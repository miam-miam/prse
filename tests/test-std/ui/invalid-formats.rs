use prse::parse;

fn main() {
    let l = "test: 5";
    let mut x = 0_i32;

    parse!("test: {x}");
    parse!(l, "test: {x}", 4);
    parse!("test: {x}",);
    parse!("test: {x}", l);
    parse!(l, 2);
}
