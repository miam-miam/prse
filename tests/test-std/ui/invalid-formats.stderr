error: expected `,`
 --> ui/invalid-formats.rs:7:5
  |
7 |     parse!("test: {x}");
  |     ^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `parse` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected token
 --> ui/invalid-formats.rs:8:26
  |
8 |     parse!(l, "test: {x}", 4);
  |                          ^

error: unexpected end of input, expected string literal
 --> ui/invalid-formats.rs:9:5
  |
9 |     parse!("test: {x}",);
  |     ^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `parse` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected string literal
  --> ui/invalid-formats.rs:10:25
   |
10 |     parse!("test: {x}", l);
   |                         ^

error: expected string literal
  --> ui/invalid-formats.rs:11:15
   |
11 |     parse!(l, 2);
   |               ^
